File: Data Structures & Algorithms/first-missing-positive/submission-2.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        int arr[100001] = {0};

        for (int num : nums) {
            if (num >= 0) {
                arr[num]++;
            }
        }

        for (int i = 1; i < 100001; i++) {
            if (arr[i] == 0) {
                return i;
            }
        }
        
    }
};

File: Data Structures & Algorithms/special-array-with-x-elements-greater-than-or-equal-x/submission-2.py
class Solution:
    def specialArray(self, nums: List[int]) -> int:
        count = [0 for i in range(1001)]

        for num in nums:
            count[num] += 1

        i = len(count) - 2

        while i >= 0:
            count[i] += count[i+1]
            if count[i] == i:
                return i
            i-=1

        return -1